File: 30_days_C#_HackerRank_Challenge/Dictionaries&Maps.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        int n = int.Parse(Console.ReadLine());
        Dictionary<string, string> phonebook = new Dictionary<string, string>();

        for (int i = 0; i < n; i++)
        {
            string[] parts = Console.ReadLine().Split();
            phonebook[parts[0]] = parts[1];
        }

        string query;
        while ((query = Console.ReadLine()) != null)
        {
            if (phonebook.ContainsKey(query))
                Console.WriteLine(query + "=" + phonebook[query]);
            else
                Console.WriteLine("Not found");
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
    }
  }
}

File: 30_days_Python_HackerRank_Challenge/Dictionaries&Maps.py
#python-2
n = int(raw_input())
phonebook = {}

for _ in range(n):
    name, number = raw_input().split()
    phonebook[name] = number

while True:
    try:
        query = raw_input()
        if query in phonebook:
            print query + "=" + phonebook[query]
        else:
            print "Not found"
    except EOFError:
        break

#python-3
n = int(input())
phonebook = {}

for _ in range(n):
    name, number = input().split()
    phonebook[name] = number

while True:
    try:
        query = input()
        print(f"{query}={phonebook[query]}" if query in phonebook else "Not found")
    except EOFError:
        break
